homeassistant/components/esphome/config_flow.py

```diff
@@ -81,56 +84,69 @@ async def async_step_discovery_confirm(self, user_input=None):
             step_id="discovery_confirm", description_placeholders={"name": self._name}
         )
 
-    async def async_step_zeroconf(self, user_input: ConfigType):
+    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
```

Comment:
This is just `user_input: ConfigType` actually (but technically they are the same defined right now).